### PR TITLE
LDEPF disturbances and faults recording: interface between the setting file reading and the SCD file

### PR DESCRIPTION
**Pull request by samirromdhani**
This TS purpose is to:

- Read the the setting file given on this Issue #256
- Offer an interface between the setting file reading and the SCD file

### CODE REVIEW

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+// SPDX-FileCopyrightText: 2023 RTE FRANCE
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.lfenergy.compas.sct.commons.dto;
+
+import org.lfenergy.compas.sct.commons.util.SettingLDEPFCsvHelper;
```
Comment (massifben) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java:7]: Interface must not have a dependence on its implementation.

Comment (samirromdhani) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java]: OK, I set no type restrictions

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * @see org.lfenergy.compas.sct.commons.util.SettingLDEPFCsvHelper
+ */
+@FunctionalInterface
+public interface LDEPFSettings<T> {
+
+    /**
+     * This method provides list of LDEPF settings
+     */
+    List<T> getSettings();
```
Comment (massifben) [inline sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java:24]: In my opinion, it is too soon to find the correct interface until the algorithm is implemented. We will probably have to refacto this interface when we implement the algorithm on the next issues.

**Review by massifben — APPROVED** [review]

File: sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java (hunk 2 above)
Comment (samirromdhani) [reply sct-commons/src/main/java/org/lfenergy/compas/sct/commons/dto/LDEPFSettings.java:24]: I don't think there is a link between the algorithm and the interface, the interface will be a parameter for the service method  to provide all the setting.

**Pull request merged**